clientd3d/draw3d.c

```diff
@@ -533,30 +533,48 @@ void SetLightingInfo(int sun_x, int sun_y, BYTE intensity)
  */
 void StretchImage(void)
 {
-  StretchC1To2(gBits,gBufferBits,area.cx,area.cy);
+  Stretch(gBits, gBufferBits, area.cx, area.cy, main_viewport_width, main_viewport_height, MAXX);
 }
 /************************************************************************/
-void StretchC1To2(BYTE *src,BYTE *dest,int width,int height)
+void Stretch(BYTE* src, BYTE* dest, int width, int height, int new_width, int new_height, int max_width)
```

Comment:
I'm not sure what this function is doing, and there's no comment on that, or what the parameters mean.  It's always doing a 2:1 stretch, in addition to some other stretching to fill the viewport? 

We want to make sure that source pixels are always drawn as squares, without anisotropy.  i.e. it would be better to reduce the vertical FOV (by restricting to the middle of the source bitmap) than to distort the image by stretching it in a non-uniform way.